packages/shared/src/components/modals/RanksModal/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactElement, useContext } from 'react';
+import classNames from 'classnames';
+import { ModalCloseButton } from '../ModalCloseButton';
+import { ResponsiveModal } from '../ResponsiveModal';
+import { useTrackModal } from '../../../hooks/useTrackModal';
+import RanksBadges from './RanksBadges';
+import RanksTags from './RanksTags';
+import DevCardFooter from './DevCardFooter';
+import IntroSection from './IntroSection';
+import { RanksModalProps } from './common';
+import AuthContext from '../../../contexts/AuthContext';
+import { ModalHeader } from '../common';
+import { Button } from '../../buttons/Button';
+import { RANKS } from '../../../lib/rank';
+
+export default function RanksModal({
+  rank,
+  progress,
+  tags,
+  hideProgress,
+  confirmationText,
+  reads,
+  devCardLimit,
+  onRequestClose,
+  className,
+  previousRank,
+  ...props
+}: RanksModalProps): ReactElement {
+  const { user, showLogin } = useContext(AuthContext);
+  useTrackModal({ isOpen: props.isOpen, title: 'ranks modal' });
+
+  return (
+    <ResponsiveModal
+      {...props}
+      onRequestClose={onRequestClose}
+      padding={false}
+      style={{ content: { maxWidth: '26.25rem' } }}
+    >
+      <ModalHeader>
+        <h3 className="font-bold typo-title3">Weekly reading goal</h3>
+        <ModalCloseButton onClick={onRequestClose} style={{ top: '0.5rem' }} />
```

Comment:
Why don't we use `top-2` class?

Reply:
It is not being applied since there's a top class inside - unable to explicitly override.

Reply:
Removed as we don't use it anymore.